compiler/rustc_parse/src/parser/attr_wrapper.rs

```diff
@@ -342,16 +342,10 @@ impl<'a> Parser<'a> {
 
         // If we support tokens at all
         if let Some(target_tokens) = ret.tokens_mut() {
-            if let Some(target_tokens) = target_tokens {
-                assert!(
```

Comment:
Why did you remove this assertion?

Reply:
It's hit in cases like `#[cfg_eval] #[rustc_dummy] 0`.

Tokens for `#[rustc_dummy] 0` are collected twice due to the new behavior of `parse_expr_force_collect`, but we still get to here despite the
```rust
        if !self.capture_cfg && matches!(ret.tokens_mut(), None | Some(Some(_))) {
            return Ok(ret);
        }
```
above because in case of `cfg_eval` the `self.capture_cfg` condition is true.